### ISSUE TITLE
Amazon Fire Stick DRM / Stalling issue on different TVs

### ISSUE THREAD

**Issue by ShanePhelan** [shaka-project/shaka-player#4551]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
3.3.10

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `main`?**
Yes

**Are you using the demo app or your own custom app?**
Custom app

**If custom app, can you reproduce the issue using our demo app?**
N/A

**What browser and OS are you using?**

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
Amazon Fire Stick model AFTMM - Android 7.1.2 - Chrome 102
Mozilla/5.0 (Linux; Android 7.1.2; AFTMM Build/NS6291; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/102.0.5005.125 Mobile Safari/537.36

**What are the manifest and license server URIs?**

Have emailed over a copy and instructions on how to get a fresh copy

**What configuration are you using?  What is the output of `player.getConfiguration()`?**

Defaults

**What did you do?**

Tried to play some Widevine encrypted dash content.

**What did you expect to happen?**
The content to playback

**What actually happened?**

We have a Samsung 2017, 2018 and 2022 TV. When the Firestick is plugged into the 2017 and 2018 devices the content plays back fine. When the Firestick is plugged into the Samsung 2022, the content stalls. 

We've seen in our data that a small percentage of users are seeing the same issue.

This is specific to our HD content that has enhanced DRM enabled ie HDCP 1.4 required. When we try the same combo of AFTMM and Samsung 2022 with our non HD content, the content plays back fine.

We're not sure if it is something wrong with the content or if there is an actual DRM issue. We don't see any errors, just a stall event:

Calling switch_(), bandwidth=7697 kbps
switch_
Stall detected at 2.812963 for 1.0329999923706055 seconds.
Seeking forward 0.1 seconds to break stall.
(all): seeked: buffered seek: presentationTime=2.912963

**Comment by ShanePhelan** [comment]
Have sent how to access our content urls to shaka-player-issues@google.com

**Comment by ShanePhelan** [comment]
Hi, can anyone provide some assistance with this issue?

**Comment by ShanePhelan** [comment]
Hi again,

here's some more info on this issue.

We've got two versions / manifests of the same video. One that uses a single adaptation set for all SD and HD renditions (we'll refer to this as single track). And one that uses an additional adaptation set (We'll refer to this as multi track).

We also have versions for both with enhanced DRM / HDCP 1.5 required. So:

Multi track with enhanced DRM
Multi track without enhanced DRM
Single track with enhanced DRM
Single track without enhanced DRM

When enhanced DRM is required, we set the video robustness level to `HW_SECURE_ALL` 

We believe the `HW_SECURE_ALL` is what is causing the playback to stall. When we try to play the non-enhanced drm multi-track or single-track, we see the same issue where it stalls after 2.5 seconds. However, if we then remove the `HW_SECURE_ALL` robustness level, the content plays back fine.

We've been testing on an AFTMM firestick and plugged into a Samsung 2022 QE43Q60BAUXXU device. We have a few other Samsung Q series and can replicate on them too. If we use a different model of Samsung ie UE32M5500, the content plays back fine with or without the `HW_SECURE_ALL` robustness level.

So it appears that even though the browser indicates that it can handle HW decryption, it is actually failing to do the HW decryption, hence why it stalls. Note, we can hear the audio for a couple of seconds but no video is rendered.
